partition_version_list.hh

```diff
@@ -24,10 +27,16 @@ public:
                 v.insert_before(*_head);
                 _tail = std::move(_head);
             }
-            _head = partition_version_ref(v);
+            _head = partition_version_ref(v, true);
+#ifdef SEASTAR_DEBUG
+            assert(!_head->is_referenced_from_entry());
+#endif
         } else {
             v.insert_after(*_tail);
-            _tail = partition_version_ref(v);
+            _tail = partition_version_ref(v, true);
```

Comment:
I don't understand the connection between the change and the bug. Declaring the ownership as unique seems like it can enable optimizations that would otherwise be disabled, not like it would change behavior.

Reply:
I also found this confusing. This seems like avoiding adding a new flag, by piggy-backing on an existing one, that happens to be unused here and can help prevent the buggy scenario. It is certainly counter-intuitive when reading the code, although the comment added above does help.

Reply:
It's just the name of the flag which is slightly confusing. The meaning of partition_version_ref::unique_owner is whether version is attached to partition entry or not.

We have this:

```
inline
bool partition_version::is_referenced_from_entry() const {
    return !prev() && _backref && !_backref->is_unique_owner();
}
```

Reply:
Ok. Please follow up with a rename.